### PR TITLE
Feature/database

### PR DESCRIPTION
**Pull request by gorbuvla**

### CODE REVIEW

File: app/src/main/java/me/gorbuvla/rssreader/RSSApp.kt [hunk 1]
```diff
@@ -2,20 +2,24 @@ package me.gorbuvla.rssreader
...
 import com.jakewharton.threetenabp.AndroidThreeTen
-import me.gorbuvla.rssreader.di.repositoryModule
-import me.gorbuvla.rssreader.di.viewModelModule
+import me.gorbuvla.articles.di.articlesModule
+import me.gorbuvla.database.di.databaseModule
+import me.gorbuvla.rss.rssModule
 import org.koin.android.ext.koin.androidContext
 import org.koin.core.context.startKoin
+import timber.log.Timber
+
```
Comment (gorbuvla) [inline app/src/main/java/me/gorbuvla/rssreader/RSSApp.kt:11]: ➖

**Comment by davidbilik** [comment]
🕵️

File: features/articles/src/main/java/me/gorbuvla/articles/model/ArticleRepository.kt [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.onEach
+import me.gorbuvla.core.database.DatabaseInteractor
+import me.gorbuvla.core.domain.Article
+import me.gorbuvla.core.rss.RssInteractor
+import timber.log.Timber
+import java.net.URL
+
+/**
+ * TODO add class description
```
Comment (gorbuvla) [inline features/articles/src/main/java/me/gorbuvla/articles/model/ArticleRepository.kt:12]: describe it

File: features/articles/src/main/java/me/gorbuvla/articles/model/ArticleRepository.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    override suspend fun fetchArticles() {
+        db.store(rss.fetch(feeds))
+    }
+
+    override fun observeArticles(): Flow<List<Article>> {
+        return db.observeArticles()
+    }
+
+    override fun observeArticle(id: Long): Flow<Article> {
+        Timber.i("ARTICLE ${id}")
```
Comment (gorbuvla) [inline features/articles/src/main/java/me/gorbuvla/articles/model/ArticleRepository.kt:42]: bu bu

File: features/feeds/build.gradle [hunk 4]
```diff
@@ -0,0 +1,37 @@
+apply plugin: 'com.android.library'
+apply from: '../../gradle/common.gradle'
+
+android {
+    buildFeatures {
+        compose true
+    }
+    composeOptions {
+        kotlinCompilerExtensionVersion "0.1.0-dev06"
```
Comment (gorbuvla) [inline features/feeds/build.gradle:9]: common version

File: features/feeds/src/main/java/me/gorbuvla/feeds/FeedsActivity.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import android.os.Bundle
+import androidx.appcompat.app.AppCompatActivity
+import androidx.ui.core.Text
+import androidx.ui.core.setContent
+import androidx.ui.layout.Column
+import androidx.ui.material.MaterialTheme
+import java.net.URL
+
+/**
+ * TODO add class description
```
Comment (gorbuvla) [inline features/feeds/src/main/java/me/gorbuvla/feeds/FeedsActivity.kt:12]: dd

File: features/feeds/src/main/java/me/gorbuvla/feeds/FeedsActivity.kt [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+        "http://android-developers.blogspot.com/atom.xml",
+        "https://swift.org/atom.xml?format=xml"
+    )
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent { 
+            MaterialTheme {
+                Column {
+
```
Comment (gorbuvla) [inline features/feeds/src/main/java/me/gorbuvla/feeds/FeedsActivity.kt:26]: fuj

File: libraries/core/src/main/java/me/gorbuvla/core/domain/Article.kt [hunk 7]
```diff
@@ -0,0 +1,14 @@
+package me.gorbuvla.core.domain
+
+import org.threeten.bp.ZonedDateTime
+
+
+/**
+ * TODO add class description
```
Comment (gorbuvla) [inline libraries/core/src/main/java/me/gorbuvla/core/domain/Article.kt:7]: td

File: libraries/rss/src/main/java/me/gorbuvla/rss/RssService.kt [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+import me.gorbuvla.core.domain.Article
+import me.gorbuvla.core.rss.RssInteractor
+import org.threeten.bp.Instant
+import org.threeten.bp.ZoneId
+import org.threeten.bp.ZonedDateTime
+import timber.log.Timber
+import java.net.URL
+
+/**
+ * TODO add class description
```
Comment (gorbuvla) [inline libraries/rss/src/main/java/me/gorbuvla/rss/RssService.kt:18]: td

File: libraries/ui/build.gradle [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+dependencies {
+    implementation "org.jetbrains.kotlin:kotlin-stdlib:$kotlin_version"
+    implementation 'androidx.core:core-ktx:1.2.0'
+    implementation 'androidx.appcompat:appcompat:1.1.0'
+    implementation 'androidx.lifecycle:lifecycle-viewmodel-ktx:2.3.0-alpha01'
+    implementation 'androidx.ui:ui-framework:0.1.0-dev06'
+    implementation 'androidx.ui:ui-layout:0.1.0-dev06'
+    implementation 'androidx.ui:ui-material:0.1.0-dev06'
+    implementation 'androidx.ui:ui-tooling:0.1.0-dev06'
```
Comment (gorbuvla) [inline libraries/ui/build.gradle:21]: common versions

**Pull request merged**